caffe2/CMakeLists.txt

```diff
@@ -310,6 +310,9 @@ if(NOT INTERN_BUILD_MOBILE OR NOT BUILD_CAFFE2_MOBILE)
       target_link_libraries(tensorpipe_agent PRIVATE torch c10d tensorpipe fmt::fmt-header-only)
       add_dependencies(tensorpipe_agent torch c10d)
       if(USE_TENSORPIPE)
+        if(USE_CUDA AND (NOT USE_ROCM))
+          target_compile_definitions(tensorpipe_agent PUBLIC USE_CUDA_NOT_ROCM)
```

Comment:
Actually, wait a second, USE_CUDA_NOT_ROCM is a macro that is only defined here, nowhere else? That might be a problem, since the CMake target `tensorpipe_agent` is _only_ used to build the C++ tests. The "real" PyTorch libraries don't use the `tensorpipe_agent` target. So I'm worried that when building the real PyTorch the macro USE_CUDA_NOT_ROCM won't be set and thus all the CUDA-specific code will be disabled.

Reply:
> That might be a problem, since the CMake target `tensorpipe_agent` is only used to build the C++ tests. The "real" PyTorch libraries don't use the `tensorpipe_agent` target.

Thanks for the catch. This is new to me. Then let me try `__HIP_PLATFORM_HCC__` instead.